File: docs/site/samples/python/examples/guides_lint-in-ci_prompts-lint-test.py
"""Wiring ``Prompt.check()`` as a CI gate under pytest.

A CI gate is a test that fails the build: load every ``*.yaml`` under a ``prompts/``
directory, construct each prompt, and assert ``check()`` returns no findings — naming
the offender otherwise. Standalone: this program first materializes a ``prompts/``
directory of shipped fixtures in a temp dir and ``chdir``s into it (a real repo keeps
its own ``prompts/`` under version control), then runs the documented parametrized test.

Run it directly (``python guides_lint-in-ci_prompts-lint-test.py``) or under pytest.
"""

from __future__ import annotations

import os
import tempfile
from pathlib import Path

import pytest
from prompting_press import Prompt, PromptingPressError

# ── Materialize the `prompts/` directory a real repo would keep under version control. ──
# A clean, shipped prompt: its untrusted-free variable needs no guard, so check() passes.
_TMP = Path(tempfile.mkdtemp(prefix="pp_lint_in_ci_"))
(_TMP / "prompts").mkdir()
(_TMP / "prompts" / "greet.yaml").write_text(
    """
name: greet
role: user
body: "Hi {{ name }}, you have {{ count }} messages."
variables:
  name:
    type: string
    trusted: true
  count:
    type: integer
    trusted: true
"""
)
os.chdir(_TMP)

# ── The CI gate itself. ──
# tests/test_prompts_lint.py  — runs under pytest
PROMPT_FILES = sorted(Path("prompts").glob("*.yaml"))


@pytest.mark.parametrize("path", PROMPT_FILES, ids=lambda p: p.name)
def test_shipped_prompt_passes_check(path: Path) -> None:
    try:
        prompt = Prompt.from_yaml(path.read_text())  # construction enforces hard invariants
    except PromptingPressError as e:
        pytest.fail(f"{path.name}: construction failed: {e}")

    report = prompt.check()
    findings = [f"{f.kind}: {f.detail}" for f in report.findings]
    assert report.passed(), f"{path.name} lint findings:\n" + "\n".join(findings)


if __name__ == "__main__":
    # Run the documented per-file check directly (no pytest runner needed).
    for _path in PROMPT_FILES:
        _prompt = Prompt.from_yaml(_path.read_text())
        _report = _prompt.check()
        assert _report.passed(), f"{_path.name} lint findings: {_report.findings}"
    print(f"prompt lint: clean — checked {len(PROMPT_FILES)} prompt(s)")

File: docs/site/samples/typescript/examples/guides_lint-in-ci_prompts-lint-test.ts
// Wiring `prompt.check()` as a CI gate under `node --test`.
//
// A CI gate is a test that fails the build: load every `*.yaml` under a `prompts/`
// directory, construct each prompt, and assert `check()` returns no findings — naming
// the offender otherwise. Standalone: this program first materializes a `prompts/`
// directory of shipped fixtures in a temp dir and `chdir`s into it (a real repo keeps
// its own `prompts/` under version control), then registers the documented test cases.
//
// Run it: `node --test guides_lint-in-ci_prompts-lint-test.ts`.

import { test } from "node:test";
import assert from "node:assert/strict";
import { mkdirSync, mkdtempSync, readdirSync, readFileSync, writeFileSync } from "node:fs";
import { tmpdir } from "node:os";
import { join } from "node:path";
import { Prompt } from "prompting-press";

// ── Materialize the `prompts/` directory a real repo would keep under version control. ──
// A clean, shipped prompt: its untrusted-free variable needs no guard, so check() passes.
const dir = mkdtempSync(join(tmpdir(), "pp-lint-in-ci-"));
mkdirSync(join(dir, "prompts"));
writeFileSync(
  join(dir, "prompts", "assistant.yaml"),
  `
name: assistant
role: system
body: "You are a support assistant for {{ company }}. Keep your replies under {{ max_words }} words."
variables:
  company:
    type: string
    trusted: true
  max_words:
    type: integer
    trusted: true
`,
);
process.chdir(dir);

// ── The CI gate itself. ──
// test/prompts-lint.test.mjs  — runs under `node --test`
for (const file of readdirSync("prompts").filter((f) => f.endsWith(".yaml"))) {
  test(`shipped prompt ${file} passes check`, () => {
    // Construction throws on the hard invariants; let that fail the test directly.
    const prompt = Prompt.fromYaml(readFileSync(`prompts/${file}`, "utf-8"));
    const report = prompt.check();
    const findings = report.findings.map((f) => `${f.kind}: ${f.detail}`);
    assert.ok(report.passed(), `${file} lint findings:\n${findings.join("\n")}`);
  });
}
